src/modules/claims/claims.service.ts

```diff
@@ -843,6 +877,198 @@ export class ClaimsService {
     return claimType.split('.roles.')[1];
   }
 
+  /**
+   * Revoke On-Chain issued claim by `claimId` or given `namespace` and `subject`. Required `claimId` or `claim` parameters.
+   *
+   * ```typescript
+   * claimsService.revokeClaim({
+   *     claim: {
+   *         namespace: 'root.roles.energyweb.iam.ewc',
+   *         subject: 'did:ethr:volta:0x00...0',
+   *     },
+   *     registrationTypes = [RegistrationTypes.OnChain, RegistrationTypes.OffChain],
+   * });
+   * ```
+   * or
+   * ```typescript
+   * claimsService.revokeClaim({
+   *     claimId: claim.id,
+   *     registrationTypes = [RegistrationTypes.OnChain, RegistrationTypes.OffChain],
+   * });
+   * ```
+   *
+   * @param {RevokeClaimOptions} options object containing options
+   * @return true if claim was revoked
+   */
+  async revokeClaim(options: RevokeClaimOptions): Promise<boolean> {
+    const revoker = this._signerService.did;
+    const { namespace, subject } = await this.getRevocationClaimDetails(
+      options
+    );
+
+    return await this._claimRevocation.revokeClaim(namespace, subject, revoker);
+  }
+
+  /**
+   * Revoke On-Chain issued claims of the given namespace for multiple subjects. Namespace must be the same for all subjects.
+   *
+   * ```typescript
+   * claimsService.revokeMultipleClaim({
+   *     claims: [{
+   *         namespace: 'root.roles.energyweb.iam.ewc',
```

Comment:
Why not move namespace from array elements to root of claims type?

Reply:
The idea behind that is to make interface simple, so you only provide `claims` or `claimIds`.